src/cluster_legacy.c

```diff
@@ -5297,10 +5298,13 @@ static int nodeExceedsHandshakeTimeout(clusterNode *node, mstime_t now) {
     return now - node->ctime > getHandshakeTimeout() ? 1 : 0;
 }
 
+#define MAX_CONNECTION_ATTEMPTS_PER_CRON 10
+#define OUTBOUND_CONN_RETRY_INTERVAL 1000
```

Comment:
Also, let's determine this value based on `cluster-node-timeout` and not a hardcoded value.